Source/Core/Common/FileUtil.cpp

```diff
@@ -516,27 +533,81 @@ bool DeleteDirRecursively(const std::string& directory)
   return success;
 }
 
-// Create directory and copy contents (optionally overwrites existing files)
-bool CopyDir(const std::string& source_path, const std::string& dest_path, const bool destructive)
+bool Copy(std::string_view source_path, std::string_view dest_path, bool overwrite_existing)
 {
+  DEBUG_LOG_FMT(COMMON, "{}: {} --> {} ({})", __func__, source_path, dest_path,
+                overwrite_existing ? "overwrite" : "preserve");
+
   auto src_path = StringToPath(source_path);
   auto dst_path = StringToPath(dest_path);
-  if (fs::equivalent(src_path, dst_path))
+  std::error_code error;
+  auto options = fs::copy_options::recursive;
+  if (overwrite_existing)
+    options |= fs::copy_options::overwrite_existing;
+  fs::copy(src_path, dst_path, options, error);
+  if (error)
+  {
+    std::error_code error_ignored;
+    if (fs::equivalent(src_path, dst_path, error_ignored))
+      return true;
+
+    ERROR_LOG_FMT(COMMON, "{}: failed {} --> {} ({}): {}", __func__, source_path, dest_path,
+                  overwrite_existing ? "overwrite" : "preserve", error.message());
+    return false;
+  }
+  return true;
+}
+
+static bool Move(const std::filesystem::path& src, const std::filesystem::path& dst,
+                 std::error_code& error)
+{
+  fs::rename(src, dst, error);
+  if (!error)
     return true;
 
-  DEBUG_LOG_FMT(COMMON, "{}: {} --> {}", __func__, source_path, dest_path);
+  // rename failed, try fallbacks
 
-  auto options = fs::copy_options::recursive;
-  if (destructive)
-    options |= fs::copy_options::overwrite_existing;
+  if (!fs::is_directory(src))
+  {
+    // src is not a directory (ie, probably a file), try to copy file + delete
+    if (!fs::copy_file(src, dst, fs::copy_options::overwrite_existing, error))
+      return false;
+    if (!fs::remove(src, error))
+      return false;
+    return true;
+  }
```

Comment:
I'm not sure if this part is actually necessary -- I originally thought it would be if source and destination are on different file systems, but std::filesystem::rename() seems to handle that just fine by itself. So I dunno, maybe this is unnecessary?